File: apps/modernization-api/src/main/java/gov/cdc/nbs/patient/contact/PatientContacts.java
package gov.cdc.nbs.patient.contact;

import java.time.Instant;
import java.util.Collection;
import gov.cdc.nbs.event.investigation.association.AssociatedWith;

public record PatientContacts(
        long patient,
        Collection<NamedByPatient> namedByPatient,
        Collection<NamedByContact> namedByContact) {

    record NamedContact(
            String local,
            String name) {

    }
    record Condition(
            String id,
            String description) {

    }
    record NamedByPatient(
            long contactRecord,
            Instant createdOn,
            Condition condition,
            NamedContact contact,
            Instant namedOn,
            String priority,
            String disposition,
            String event,
            Investigation associatedWith) {

    }


    record NamedByContact(
            long contactRecord,
            Instant createdOn,
            Condition condition,
            NamedContact contact,
            Instant namedOn,
            String event,
            AssociatedWith associatedWith) {

    }


    public record Investigation(
            long id,
            String local,
            String condition) {
    }
}

File: apps/modernization-ui/src/patient/profile/contact/PatientContacts.ts
export type Contact = {
    id: string;
    name: string;
};

export type AssociatedWith = {
    id: string;
    condition: string;
    local: string;
};

export type Tracing = {
    contactRecord: string;
    createdOn: any;
    condition?: string | null;
    namedOn: any;
    priority?: string | null;
    disposition?: string | null;
    event: string;
    contact: Contact;
    associatedWith?: AssociatedWith | null;
};
